rtrlib/lib/aspa_vector.h

```diff
@@ -0,0 +1,94 @@
+/*
+* This file is part of RTRlib.
+*
+* This file is subject to the terms and conditions of the MIT license.
+* See the file LICENSE in the top level directory for more details.
+*
+* Website: http://rtrlib.realmv6.org/
+*/
+
+#ifndef LRTR_VECTOR_UTILS_H
+#define LRTR_VECTOR_UTILS_H
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+/**
+ * @brief ASPA Object
+ * here the customer as number (CAS) authorizes a set of provider AS numbers
+ */
+struct aspas_record {
+	uint32_t customer_asn;
+	size_t provider_count;
+	uint32_t *provider_asns;
+};
+
+/**
+ * @brief Struct which is similar in function to std::vector from C++.
+ * If the vector is running full a larger chunk of memory is allocated and the data is copied over.
+ */
+struct aspa_vector {
+	uint32_t size;
+	size_t capacity;
+	struct aspas_record *data;
+};
+/**
+ * creates an vector object
+ * @param[vector_pointer] the pointer to the newly created pointer will be written to *vector_pointer
+ * @result Valid pointer to an aspa_vector struct
+ * @result Null On error.
+ */
+int aspa_vector_create(struct aspa_vector **vector_pointer);
+
+/**
+ * deletes the given vector
+ * @param[vector] aspa_vector which will be deleted
+ * @result 0 On success.
+ * @result -1 On error.
+ */
+int aspa_vector_delete(struct aspa_vector *vector);
+
+/**
+ * reallocates the vector to increase its size
+ * @param[vector] aspa_vector which should get an increase in size
+ * @result 0 On success.
+ * @result -1 On error.
+ */
+int aspa_vector_reallocate(struct aspa_vector *vector);
+
+/**
+ * sorts the given list of aspas records by the customer as number (CAS) in ascending order
```

Comment:
please update the description
- inserts in correct order
- assumed there to be enough capacity (else it might cause buffer overflow)